apps/emqx/src/emqx_tls_lib.erl

```diff
@@ -576,7 +578,7 @@ to_client_opts(Type, Opts) ->
                     {ciphers, Ciphers},
                     {reuse_sessions, Get(reuse_sessions)},
                     {depth, Get(depth)},
-                    {password, ensure_str(Get(password))},
+                    {password, ensure_password(Get(password))},
```

Comment:
Q: Why not mark it as `emqx_schema_secret:secret()` on the schema level, and then call simple `unwrap/1` here?

Reply:
fixed